src/test/scala/au/com/onegeek/sbtdotenv/SbtDotenvSpec.scala

```diff
@@ -38,45 +38,79 @@ class SbtDotenvSpec extends WordSpec with Matchers {
-      SbtDotenv.parseLine("F.OO=bar") should equal(None)
+    "accept lines with quoted variables and strips quotes" in {
+      SbtDotenv.parse("FOO='a=b==ccddd'") should equal(Map("FOO" -> "a=b==ccddd"))
+      SbtDotenv.parse("FOO=\"blah # blah \r blah \n blah \"") should equal(Map("FOO" -> "blah # blah \r blah \n blah "))
+    }
+
+    "accept lines with whitespace around assignment operator" in {
+      SbtDotenv.parse("FOO   =   boo") should equal(Map("FOO" -> "boo"))
+    }
 
-      SbtDotenv.isValidLine("1234=5678") should equal(false)
-      SbtDotenv.parseLine("1234=5678") should equal(None)
+    "accept lines with escaped characters and unescape them" in {
+      SbtDotenv.parse("FOO=' \\\' \\\' '") should equal(Map("FOO" -> " \' \' "))
+    }
 
-      SbtDotenv.isValidLine("COVERALLS_REPO_TOKEN=NTHnTHSNthnTHSntNt09aoesNTH6") should equal(true)
-      SbtDotenv.parseLine("COVERALLS_REPO_TOKEN=NTHnTHSNthnTHSntNt09aoesNTH6") should equal(Some("COVERALLS_REPO_TOKEN", "NTHnTHSNthnTHSntNt09aoesNTH6"))
+    "accept lines with leading whitespace before variable name" in {
+      SbtDotenv.parse("   FOO=noo") should equal(Map("FOO" -> "noo"))
+    }
 
-      SbtDotenv.isValidLine("SOMETHING_TOKEN=I love kittens") should equal(true)
-      SbtDotenv.parseLine("SOMETHING=I love kittens") should equal(Some("SOMETHING", "I love kittens"))
+    "accept lines with leading export and ignore the export" in {
+      SbtDotenv.parse(" export FOO=noo") should equal(Map("FOO" -> "noo"))
+    }
 
-      SbtDotenv.isValidLine("FOO='a=b==ccddd'") should equal(true)
-      SbtDotenv.parseLine("FOO='a=b==ccddd'") should equal(Some("FOO", "'a=b==ccddd'"))
+    "accept multi-line variables" in {
+      val content = """MY_CERT="-----BEGIN CERTIFICATE-----
+        |123456789qwertyuiopasdfghjklzxcvbnm
+        |-----END CERTIFICATE-----
+        |"
+      """.stripMargin
+      SbtDotenv.parse(content) should equal(Map("MY_CERT" -> """-----BEGIN CERTIFICATE-----
+                                                             |123456789qwertyuiopasdfghjklzxcvbnm
+                                                             |-----END CERTIFICATE-----
+                                                             |""".stripMargin))
+    }
 
-      SbtDotenv.isValidLine("F") should equal(false)
-      SbtDotenv.parseLine("F") should equal(None)
+    "validate correct lines in a .env file" in {
+      SbtDotenv.parse("FOO=bar") should equal(Map("FOO" -> "bar"))
 
-      SbtDotenv.isValidLine("") should equal(false)
-      SbtDotenv.parseLine("") should equal(None)
+      SbtDotenv.parse("FOO=1234") should equal(Map("FOO" -> "1234"))
 
-      SbtDotenv.isValidLine("WITHOUT_COMMENT=ThisIsValue # here is a comment") should equal(true)
-      SbtDotenv.parseLine("WITHOUT_COMMENT=ThisIsValue # here is a comment") should equal(Some("WITHOUT_COMMENT", "ThisIsValue"))
+      SbtDotenv.parse("F.OO=bar") should equal(Map())
```

Comment:
this goes against https://github.com/mefellows/sbt-dotenv/pull/24

Reply:
I'm not a fan of dots as they're not valid shell identifiers:
```
export M.OO=moo; echo ${M.OO}
-bash: export: `M.OO=moo': not a valid identifier
-bash: ${M.OO}: bad substitution
```

Having said that, happy to bring them in.

Reply:
Hrm, actually dots work fine with `getenv`/`setenv` so it's just bash that's being picky here:
```
$ ruby -e "ENV['M.OO'] = 'moo'; puts ENV['M.OO']"
```

Reply:
Yeah, my initial reaction was the same until I looked up the spec.